### PR TITLE
feat: notification channels + pattern-based denial alerting

### PR DESCRIPTION
**Pull request by marcelogdeandrade**
## Summary

Adds a denial alerting system to CrabTrap. When a bot is denied access to a new URL pattern, its managers are immediately notified via their configured notification channels. A separate periodic digest uses an LLM to summarize all recent denials into an actionable overview.

### What this PR adds

**1. Notification Channels (CRUD)**

Managers configure where they receive alerts — currently Slack (via bot token). The `Sender` interface is extensible for other backends.

- DB table: `notification_channels` (owner_id, bot_id, channel_type, destination, enabled)
- Admin API: `GET/POST/PUT/DELETE /admin/notification-channels`, `POST .../test`
- Web UI: "Notification Channels" section on bot detail page (add, remove, toggle, test)
- Auth: managers manage channels for their bots; admins for all

**2. Immediate Denial Alerts**

Each new denied URL pattern triggers an immediate notification. Same pattern within cooldown is silent (deduped).

- Plugs into the existing `notifications.Dispatcher` as a `Channel` implementation
- Pattern normalization: host + first 2 path segments (strips query/fragment/default ports)
- GraphQL-aware: extracts `operationName` from request body for `/graphql` endpoints
- In-memory dedup with DB persistence for multi-instance safety
- DB table: `denial_notifications` (bot_id, method, url_pattern, cooldown_until, digested_at)

**3. Periodic LLM Digest (optional)**

If an LLM adapter is configured, a background job runs hourly (configurable) and sends a summary of all new denials per bot. Only patterns not yet included in a previous digest are reported — prevents repeating the same information.

- Uses the existing fast LLM adapter
- Groups all un-digested denials per bot → LLM prompt → 1-2 sentence summary
- Marks patterns as digested after sending

### Flow

```
Immediate alert (per new pattern):
  Proxy DENY → audit logger → dispatcher → alerting.Service.Notify()
    → normalize URL → dedup check → async: record + resolve managers + send

Periodic digest (hourly, only if LLM configured):
  digestLoop tick → query un-digested denial_notifications
    → group by bot → LLM summarize → send → mark digested
```

### Config

```yaml
alerting:
  enabled: true
  cooldown: 1h            # dedup window per (bot, pattern)
  digest_interval: 1h     # LLM digest frequency (only runs if LLM is available)
  slack:
    bot_token: "${CRABTRAP_SLACK_BOT_TOKEN}"
```

### Extending with new channel types

```go
type Sender interface {
    Send(ctx context.Context, destination string, msg Message) error
}
// Register at startup:
alertService.RegisterSender("pagerduty", myPagerDutySender)
// Users create channels with:
// {"channel_type": "pagerduty", "destination": "routing-key-xxx"}
```

### Files

| Area | Files |
|------|-------|
| Migration | `internal/db/migrations/004_notification_channels.sql` |
| Core logic | `internal/alerting/alerting.go` (service), `pattern.go`, `store.go`, `sender.go`, `summarizer.go` |
| Admin API | `internal/admin/notification_handlers.go`, `api.go` (routes + struct fields) |
| Config | `internal/config/config.go` |
| Wiring | `cmd/gateway/main.go` |
| Frontend | `web/src/components/UsersPanel.tsx`, `web/src/api/client.ts`, `web/src/types/index.ts` |
| Docs | `docs/alerting.md`, `README.md` |

## Test plan

- [x] Pattern normalization: 17 unit tests (REST, GraphQL, edge cases)
- [x] All existing tests pass (16 packages)
- [x] TypeScript compiles (`tsc --noEmit`)
- [x] Security: Slack mrkdwn escaping, URL escaping in links
- [x] Race prevention: cooldown set before async dispatch
- [x] Graceful shutdown: Stop() terminates background loops
- [x] Digest dedup: digested_at prevents repeated digests
- [ ] Integration test: channel CRUD + auth enforcement
- [ ] Integration test: denial → immediate notification with mock sender
- [ ] Integration test: digest with mock LLM
- [ ] Manual: Slack app setup, trigger denials, verify messages

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/alerting.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+3. Install the app to your workspace
+4. Copy the Bot User OAuth Token (starts with `xoxb-`)
+5. Set it as `CRABTRAP_SLACK_BOT_TOKEN` in your environment
+6. Invite the bot to any channels where you want alerts: `/invite @CrabTrap`
+
+## Managing Notification Channels
+
+Managers configure where their bot alerts go via the admin API:
+
+```bash
```
Comment (bjhaid) [inline docs/alerting.md:41]: given managers are not admin should the url be hosted under admin?

File: docs/alerting.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  -H "Authorization: Bearer $TOKEN"
+
+# List your channels
+curl http://localhost:8081/admin/notification-channels \
+  -H "Authorization: Bearer $TOKEN"
+```
+
+## URL Pattern Normalization
+
+URLs are grouped by host and the first two path segments. Query parameters, fragments, and default ports are stripped:
```
Comment (bjhaid) [inline docs/alerting.md:65]: what happens with denials for something a graphql request? where the payload is what determines what should be allowed/denied?

Comment (marcelogdeandrade) [reply docs/alerting.md:65]: I've implemented an improvement [here](https://github.com/brexhq/CrabTrap/pull/20/commits/17cd823d7bdf6207f7ec50fc1e0faae6bf94623e) (now this is actually me talking, not claude lol)
